src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -56,71 +67,73 @@ int main(int argc, char* argv[])
     }
     phAudioFile->getFileSpec(stFileSpec);
 
-    //////////////////////////////////////////////////////////////////////////////
-    // open the output text file
-    hOutputFile.open(sOutputFilePath.c_str(), std::ios::out);
-    if (!hOutputFile.is_open())
+    hOutputFile->openFile(sOutputFilePath, CAudioFileIf::kFileWrite, &stFileSpec);
+    if (!hOutputFile->isOpen())
     {
-        cout << "Text file open error!";
-        CAudioFileIf::destroy(phAudioFile);
+        cout << "Wave file open error!";
+        CAudioFileIf::destroy(hOutputFile);
         return -1;
     }
 
+
+    //////////////////////////////////////////////////////////////////////////////
+//  Init Filter
+    pcCombFilter->init(CCombFilterIf::kCombFIR, 0.5, stFileSpec.fSampleRateInHz, stFileSpec.iNumChannels);
```

Comment:
From my understanding, to use init function, you should pass the (delay*samplerate) as the second parameter, instead of using a fix number.